CMakeLists.txt

```diff
@@ -1,4 +1,5 @@
-cmake_minimum_required(VERSION 3.12.0)
+# CMake 3.16 introduced support for unity builds but 3.17.2 is more stable
+cmake_minimum_required(VERSION 3.17.2)
```

Comment:
This change requires that CMake 3.17.2 (or newer) is bundled with the build instead of relying on the host system

Reply:
cmake has an ubuntu apt repo, so that's easy for CI. With a cmake version guard, best of both worlds.